cmake/zlib.cmake

```diff
@@ -0,0 +1,30 @@
+set(ZLIB_DIR ${CMAKE_CURRENT_BINARY_DIR}/_deps/zlib-1.1.4-src/ZLib)
+
+FetchContent_Populate(zlib DOWNLOAD_EXTRACT_TIMESTAMP
+    GIT_REPOSITORY https://github.com/TheSuperHackers/zlib-1.1.4
+    GIT_TAG        ac753eee3990a2f592bd4807ff0b30ff572c2104
+    SOURCE_DIR     ${ZLIB_DIR}
+)
+    
+add_library(libzlib STATIC)
```

Comment:
I would suggest changing this to mark it as EXCLUDE_FROM_ALL as per https://stackoverflow.com/questions/68671224/how-to-use-exclude-from-all-in-cmake so that the library isn't built by default unless its a dependency of something else that is getting built.

Maybe even just gate the inclusion of the cmake behind ZLIB not being found so if its provided either by a package manager or as part of the OS the older zlib isn't even downloaded?

Reply:
Fixed